examples/fluids/qfunctions/newtonian.h

```diff
@@ -324,12 +313,21 @@ CEED_QFUNCTION(IFunction_Newtonian)(void *ctx, CeedInt Q,
                                     dXdx[j][2] * Flux[k][2]);
 
     const CeedScalar body_force[5] = {0, s.U.density *g[0], s.U.density *g[1], s.U.density *g[2], 0};
-    for (CeedInt j=0; j<5; j++)
-      v[j][i] = wdetJ * (q_dot[j][i] - body_force[j]);
 
     // -- Stabilization method: none (Galerkin), SU, or SUPG
     CeedScalar Tau_d[3], stab[5][3], U_dot[5] = {0};
-    for (CeedInt j=0; j<5; j++) U_dot[j] = q_dot[j][i];
+    if (context->use_primitive) {
+      CeedScalar qi_dot[5], dx0[3] = {0};
+      for (int j=0; j<5; j++) qi_dot[j] = q_dot[j][i];
+      State s_dot = StateFromY_fwd(context, s, qi_dot, x_i, dx0);
+
+      UnpackState_U(s_dot.U, U_dot);
+    } else {
+      for (CeedInt j=0; j<5; j++) U_dot[j] = q_dot[j][i];
+    }
```

Comment:
Should we use `StateFromQi_fwd` here for uniformity? I think the "cost" is negligible in the scheme of things, and should be optimized out after inlining.

Reply:
I had the same thought going to sleep last night. haha. Yeah, after inlining, the conservative case should effectively break down to copying data into the `State`, then copying it back out.